google/cloud/spanner_dbapi/connection.py

```diff
@@ -198,9 +199,13 @@ def commit(self):
         if self._autocommit:
             warnings.warn(AUTOCOMMIT_MODE_WARNING, UserWarning, stacklevel=2)
         elif self._transaction:
-            self._transaction.commit()
-            self._release_session()
-            self._statements = []
+            try:
+                self._transaction.commit()
+                self._release_session()
+                self._statements = []
+            except Aborted:
+                self.retry_transaction()
+                self.commit()
```

Comment:
The retry attempt itself + the `commit` call should all be placed inside a retry-loop, as the retry attempt itself could also abort, and should then also be retried.

Reply:
Retry inside retry? Okay)

Reply:
> Retry inside retry? Okay)

Yep. I haven't looked into the other retry implementation yet, but you might need to make this a little bit more generic. Any statement during a retry could of course also return an `Aborted` error, and any of those should also automatically restart the retry.